Fix sub-man installation on Alma and Rocky 8

bocekm opened the pull request:
If the python3-syspurpose package is installed on the system prior to the conversion, it has a different NEVRA than subscription-manager requires. The subscription-manager is strict about it - the syspurpose VRA (version, release, arch) needs to match exactly the sub-man VRA. And because we're downloading the sub-man packages from UBI, the VRA doesn't match.

Previously we thought this issue affects just CentOS Linux 8 but it does affect Alma and Rocky 8 as well.

Follows-up on #869.

abadger reviewed (approved):
Works for me.

The only drawback if this captures too many distros is that on rollback, the system could have the python-syspurpose and json-c packages from UBI instead of (an older version) from the original vendor.

bocekm commented:
I've manually verified it works on Alma 8.8 and OL 8.latest (8.8).
Alma 8.8 had python3-syspurpose and json-c pre-installed and both got downloaded and updated together with installation of sub-man.
OL 8.latest had json-c pre-installed and it got downloaded and updated together with installation of sub-man.
